sympy/core/numbers.py

```diff
@@ -1101,37 +1100,22 @@ def __int__(self):
         return int(mlib.to_int(self._mpf_))  # uses round_fast = round_down
 
     def __eq__(self, other):
-        from sympy.logic.boolalg import Boolean
-        try:
-            other = _sympify(other)
-        except SympifyError:
-            return NotImplemented
-        if isinstance(other, Boolean):
-            return False
-        if other.is_NumberSymbol:
-            if other.is_irrational:
-                return False
-            return other.__eq__(self)
-        if other.is_Float:
-            # comparison is exact
-            # so Float(.1, 3) != Float(.1, 33)
-            return self._mpf_ == other._mpf_
-        if other.is_Rational:
-            return other.__eq__(self)
-        if other.is_Number:
-            # numbers should compare at the same precision;
-            # all _as_mpf_val routines should be sure to abide
-            # by the request to change the prec if necessary; if
-            # they don't, the equality test will fail since it compares
-            # the mpf tuples
-            ompf = other._as_mpf_val(self._prec)
-            return bool(mlib.mpf_eq(self._mpf_, ompf))
-        if not self:
-            return not other
-        return False    # Float != non-Number
+        if isinstance(other, float):
+            other = Float(other)
+        return Basic.__eq__(self, other)
 
     def __ne__(self, other):
-        return not self == other
+        eq = self.__eq__(other)
```

Comment:
What is the reason for this over the standard __ne__ implementation?

Reply:
I didn't write this but judging from the code one must handle the NotImplementedError. And rather than duplicate what is done in  `__eq__` we just call `__eq__` and return the the `not` of it.

Reply:
@oscarbenjamin , do you have any feedback for this?

Reply:
BTW, `not NotImplementedError` is still NotImplementedError so we could return `not self.__eq__(other)` -- is that more what you expect to see, @ThePauliPrinciple ?